cfgov/unprocessed/apps/youth-employment-success/js/reducers/route-option-reducer.js

```diff
@@ -26,6 +28,46 @@ const updateDailyCostAction = actionCreator(
   actionTypes.UPDATE_DAILY_COST
 );
 
+/**
+ *
+ * @param {object} state the current state of this reducer
+ * @param {number} index the index of the route we want to retrieve
+ * @returns {object} the route object at the desired index
+ */
+function routeSelector( state, index ) {
+  return state.routes[index] || {};
+}
+
+/**
+ *
+ * @param {array} routes Route options the user has created
+ * @param {object} nextRoute a route model object
+ * @returns {array} the array of routes with the most recent route added
+ */
+function addRouteOption( routes, nextRoute ) {
+  return routes.concat( nextRoute );
+}
+
+/**
+ *
+ * @param {array} routes the array of routes in the reducer's state
+ * @param {number} routeIndex the index of the route object we want to update
+ * @param {object} data object describing the updates we want to make
+ *  to the route object's proeprties
+ * @returns {object} the next state of the reducer
+ */
+function updateRouteData( routes, routeIndex, data ) {
+  return {
```

Comment:
If you have a big block of logic you're returning I might store it as an object and then return that, which I think is a little clearer when parsing over the code, like:

```
const nextState = {
  routes: routes.map( ( route, index ) => {
    if ( index !== routeIndex ) {
      return route;
    }

    return assign( route, data );
  } )
};

return nextState;
```